sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -0,0 +1,730 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_iot_pnp_client.h
+ *
+ * @brief Definition for the Azure IoT PnP device SDK.
+ *
+ * @warning THIS LIBRARY IS IN PREVIEW. APIS ARE SUBJECT TO CHANGE UNTIL GENERAL AVAILABILITY.
+ */
+
+#ifndef _az_IOT_PNP_CLIENT_H
+#define _az_IOT_PNP_CLIENT_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/iot/az_iot_hub_client.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Azure IoT PnP Client options.
+ *
+ */
+typedef struct
+{
+  az_span module_id; /**< The module name (if a module identity is used). */
+  az_span user_agent; /**< The user-agent is a formatted string that will be used for Azure IoT
+                         usage statistics. */
+  az_span* component_names; /**< The array of component names for this device. */
+  int32_t
+      component_names_length; /**< The number of component names in the `component_names` array. */
```

Comment:
nit: how about `number_of_components`
```suggestion
      number_of_components; /**< The length of the \p component_names array of spans (i.e. the number of components to look for). */
```